binders/pulsar-binder/spring-cloud-stream-binder-pulsar/src/main/java/org/springframework/cloud/stream/binder/pulsar/PulsarBinderUtils.java

```diff
@@ -91,65 +100,70 @@ static Map<String, Object> mergeModifiedProducerProperties(ProducerConfigPropert
 		// Combine both base and extended layers
 		var layeredConsumerProps = new HashMap<>(layeredBaseConsumerProps);
 		layeredConsumerProps.putAll(layeredExtConsumerProps);
 		return layeredConsumerProps;
 	}
 
 	/**
-	 * Merges properties defined at the binder and binding level (binding properties
-	 * override binder properties).
+	 * Merges properties defined at the binder and binding level with binding properties
+	 * overriding binder properties.
 	 * <p>
-	 * <b>NOTE:</b> Properties whose value is not different from the default value in the
-	 * {@code baseProps} are not included in the merged result.
-	 * @param baseProps the map of base level properties (eg. 'spring.pulsar.consumer.*')
+	 * @param globalProps the map of global level properties (eg. 'spring.pulsar.consumer.*')
 	 * @param binderProps the map of binder level properties (eg.
 	 * 'spring.cloud.stream.pulsar.binder.consumer.*')
 	 * @param bindingProps the map of binding level properties (eg.
 	 * 'spring.cloud.stream.pulsar.bindings.myBinding-in-0.consumer.*')
-	 * @return map of merged binder and binding properties including only properties whose
-	 * value has changed from the same property in the base properties
+	 * @param includeDefaults whether to also include unmodified properties with their default values
+	 * @return map of merged binder and binding properties with binding properties overriding binder properties
 	 */
-	static Map<String, Object> mergePropertiesWithPrecedence(Map<String, Object> baseProps,
-			Map<String, Object> binderProps, Map<String, Object> bindingProps) {
-		Objects.requireNonNull(baseProps, "baseProps must be specified");
+	static Map<String, Object> mergePropertiesWithPrecedence(Map<String, Object> globalProps,
+			Map<String, Object> binderProps, Map<String, Object> bindingProps, boolean includeDefaults) {
+		Objects.requireNonNull(globalProps, "globalProps must be specified");
 		Objects.requireNonNull(binderProps, "binderProps must be specified");
 		Objects.requireNonNull(bindingProps, "bindingProps must be specified");
 
-		Map<String, Object> newOrModifiedBinderProps = extractNewOrModifiedProperties(binderProps, baseProps);
+		Map<String, Object> newOrModifiedBinderProps = extractNewOrModifiedProperties(binderProps, globalProps);
 		LOGGER.trace(() -> "New or modified binder props: %s".formatted(newOrModifiedBinderProps));
 
-		Map<String, Object> newOrModifiedBindingProps = extractNewOrModifiedProperties(bindingProps, baseProps);
+		Map<String, Object> newOrModifiedBindingProps = extractNewOrModifiedProperties(bindingProps, globalProps);
 		LOGGER.trace(() -> "New or modified binding props: %s".formatted(newOrModifiedBindingProps));
 
 		Map<String, Object> mergedProps = new HashMap<>(newOrModifiedBinderProps);
 		mergedProps.putAll(newOrModifiedBindingProps);
-		LOGGER.trace(() -> "Final merged props: %s".formatted(mergedProps));
 
+		// Add in default properties for any props not customized by the user
```

Comment:
Ultimately this is the "fix" in that it ensures that all extended properties values are passed into the object setters.